diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..c102d48
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,7 @@
+rpm:
+	tar czpvf /tmp/libprop.tar.gz --exclude SOURCES ../portableproplib
+	mkdir -p SOURCES && mv /tmp/libprop.tar.gz SOURCES/
+	rpmbuild -ba --define "_topdir ${PWD}" SPECS/libprop.spec
+
+clean:
+	rm -rf BUILD BUILDROOT RPMS SOURCES SRPMS
diff --git a/SPECS/libprop.spec b/SPECS/libprop.spec
new file mode 100644
index 0000000..ecf5819
--- /dev/null
+++ b/SPECS/libprop.spec
@@ -0,0 +1,38 @@
+Name:		libprop
+Version:	0.6.5
+Release:	1%{?dist}
+Summary:	Portable Property container object library
+Group:		System Environment/Libraries
+License:	BSD
+URL:		https://github.com/xtraeme/portableproplib
+Source0:	libprop.tar.gz
+
+BuildRequires:	automake
+BuildRequires:	autoconf
+BuildRequires:	libtool
+BuildRequires:	zlib-devel
+
+%description
+The proplib library provides an abstract interface for creating and
+manipulating property lists. Property lists have object types for boolean
+values, opaque data, numbers, and strings. Structure is provided by the
+array and dictionary collection types.
+
+%prep
+%setup -q -n portableproplib
+
+%build
+ln -s README.md README
+./bootstrap
+./configure --prefix=/usr --libdir=/usr/lib64
+make %{?_smp_mflags}
+
+%install
+make install DESTDIR=%{buildroot}
+
+%files
+%{_libdir}/*
+%{_includedir}/prop/*
+%{_mandir}/man3/*
+
+%changelog
diff --git a/configure.ac b/configure.ac
index 984ddff..68acde7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -28,7 +28,7 @@ LT_REVISION=13
 ## increment if any interfaces have been added; set to 0
 ## if any interfaces have been changed or removed. removal has
 ## precedence over adding, so set to 0 if both happened.
-LT_AGE=1
+LT_AGE=0
 
 AC_SUBST(LT_CURRENT)
 AC_SUBST(LT_REVISION)
diff --git a/src/prop_number.c b/src/prop_number.c
index daf035c..4434fbf 100644
--- a/src/prop_number.c
+++ b/src/prop_number.c
@@ -192,7 +192,7 @@ _prop_number_externalize(struct _prop_object_externalize_context *ctx,
 	 * we output in decimal.
 	 */
 	if (pn->pn_value.pnv_is_unsigned)
-		snprintf(tmpstr, sizeof(tmpstr), "%" PRIu64,
+		snprintf(tmpstr, sizeof(tmpstr), "0x%" PRIx64,
 		    pn->pn_value.pnv_unsigned);
 	else
 		snprintf(tmpstr, sizeof(tmpstr), "%" PRIi64,
